src/unix/linux_like/linux/gnu/b64/powerpc64/mod.rs

```diff
@@ -10,8 +10,12 @@ pub type nlink_t = u64;
 pub type blksize_t = i64;
 pub type suseconds_t = i64;
 pub type __u64 = ::c_ulong;
+pub type gregset_t = [::c_ulong; 48];
+pub type fpregset_t = [::c_double; 33];
 
 s! {
+    pub struct pt_regs {}
```

Comment:
The `s!` macro does not regconise fieldless struct. I was thinking about declaring this struct without using the macro. However I do not think it is a good idea because I would have to bring all the guts that `s!` macro introduce manually to the struct. I would like to ask the core team for suggestion to work-around this shortcoming. Many thanks in advance

Reply:
I was thinking of adding fieldless struct format for `s!` macro:

```diff
diff --git a/src/macros.rs b/src/macros.rs
index b314f60ff..6de765e35 100644
--- a/src/macros.rs
+++ b/src/macros.rs
@@ -70,6 +70,21 @@ macro_rules! s {
     (it: $(#[$attr:meta])* pub union $i:ident { $($field:tt)* }) => (
         compile_error!("unions cannot derive extra traits, use s_no_extra_traits instead");
     );
+    (it: $(#[$attr:meta])* pub struct $i:ident {}) => (
+        __item! {
+            #[repr(C)]
+            #[cfg_attr(feature = "extra_traits", derive(Debug, Eq, Hash, PartialEq))]
+            #[allow(deprecated)]
+            $(#[$attr])*
+            pub struct $i;
+        }
+        #[allow(deprecated)]
+        impl ::Copy for $i {}
+        #[allow(deprecated)]
+        impl ::Clone for $i {
+            fn clone(&self) -> $i { *self }
+        }
+    );
     (it: $(#[$attr:meta])* pub struct $i:ident { $($field:tt)* }) => (
         __item! {
             #[repr(C)]
```

However it does not change the fact that fieldless struct would cause discrepancies in size and align which lead to test failures:

```
bad pt_regs size: rust: 0 (0x0) != c 352 (0x160)
bad pt_regs align: rust: 1 (0x1) != c 8 (0x8)
size of struct pt_regs is 352 in C and -980681736 in Rust
```

Any guidance / example on how to handle this type of C struct is greatly appreciated